contracts/schemes/Competition.sol

```diff
@@ -0,0 +1,349 @@
+        /** get how many elements are greater than a given element*/
+        for (uint256 i; i < topSuggestionsOrdered.length; i++) {
+            if (suggestions[topSuggestionsOrdered[i]].totalVotes > suggestions[_suggestionId].totalVotes) {
+                index++;
+            }
+        }
+    }
+
+    /**
+    * @dev refreshTopSuggestions this function maintain a winners list array.
+    * @param _proposalId proposal id
+    * @param _suggestionId suggestion id
+    */
+    function refreshTopSuggestions(bytes32 _proposalId, uint256 _suggestionId) private {
+        uint256[] storage topSuggestionsOrdered = proposals[_proposalId].topSuggestionsOrdered;
+        if (topSuggestionsOrdered.length < proposals[_proposalId].numberOfWinners) {
+            topSuggestionsOrdered.push(_suggestionId);
+        } else {
+         /** get the index of the smallest element **/
+            uint256 smallest = 0;
+            for (uint256 i; i < proposals[_proposalId].numberOfWinners; i++) {
+                if (suggestions[topSuggestionsOrdered[i]].totalVotes <
+                    suggestions[topSuggestionsOrdered[smallest]].totalVotes) {
+                    smallest = i;
+                }
+            }
+
+            if (suggestions[topSuggestionsOrdered[smallest]].totalVotes < suggestions[_suggestionId].totalVotes) {
+                topSuggestionsOrdered[smallest] = _suggestionId;
+            }
+        }
+    }
+
+    /**
+    * @dev redeem a winning suggestion reward
+    * @param _suggestionId suggestionId
+    * @param _beneficiary - the reward beneficiary
+    */
+    function _redeem(uint256 _suggestionId, address payable _beneficiary) private {
+        bytes32 proposalId = suggestions[_suggestionId].proposalId;
+        require(proposalId != bytes32(0), "suggestion not exist");
+        Proposal storage proposal = proposals[proposalId];
+        // solhint-disable-next-line not-rely-on-time
+        require(proposal.endTime < now, "competition is still on");
+        uint256 amount;
+        //check if there is a win
+        for (uint256 i = 0; i < proposal.topSuggestionsOrdered.length; i++) {
+            if (suggestions[proposal.topSuggestionsOrdered[i]].suggester != address(0)) {
+                uint256 orderIndex = getOrderedIndexOfSuggestion(proposalId, _suggestionId);
+                suggestions[_suggestionId].suggester = address(0);
+                uint256 rewardPercentage = 0;
+                uint256 numberOfTieSuggestions = proposal.tiesSuggestions[suggestions[_suggestionId].totalVotes];
+                uint256 j;
+                //calc the reward percentage for this suggestion
+                for (j = orderIndex; j < (orderIndex+numberOfTieSuggestions); j++) {
+                    rewardPercentage = rewardPercentage.add(proposal.rewardSplit[j]);
+                }
+                rewardPercentage = rewardPercentage.div(numberOfTieSuggestions);
```

Comment:
you may get some rounding errors here and distribute too litte. (i.e. if the reward percentage is 5 numberofTies is 3, each will getonly 1 percent.

Reply:
we need to see what to do with this dust .
- send it back to the dao ?
-give it to the winner?

fixing the rounding issue might add some unneeded complexity

Reply:
added sendLeftOverFubnds at 
456cbff